DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/AddDelegateToGroupController.cs

```diff
@@ -0,0 +1,151 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.TrackingSystem.Delegates
+{
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.DataServices.UserDataService;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Attributes;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Helpers.FilterOptions;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.AddDelegateToGroup;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.FeatureManagement.Mvc;
+
+    [FeatureGate(FeatureFlags.RefactoredTrackingSystem)]
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [SetDlsSubApplication(nameof(DlsSubApplication.TrackingSystem))]
+    [SetSelectedTab(nameof(NavMenuTab.Delegates))]
+    [Route("TrackingSystem/Delegates/Groups/AddDelegateToGroup")]
+    public class AddDelegateToGroupController : Controller
+    {
+        private const string AddDelegateToGroupCookieName = "AddDelegateToGroupFilter";
+        private readonly CentreCustomPromptHelper centreCustomPromptHelper;
+        private readonly IClockService clockService;
+        private readonly IGroupsDataService groupsDataService;
+        private readonly IGroupsService groupsService;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserDataService userDataService;
+
+        public AddDelegateToGroupController(
+            IJobGroupsDataService jobGroupsDataService,
+            IUserDataService userDataService,
+            CentreCustomPromptHelper centreCustomPromptHelper,
+            IGroupsService groupsService,
+            IGroupsDataService groupsDataService,
+            IClockService clockService
+        )
+        {
+            this.centreCustomPromptHelper = centreCustomPromptHelper;
+            this.jobGroupsDataService = jobGroupsDataService;
+            this.userDataService = userDataService;
+            this.groupsService = groupsService;
+            this.groupsDataService = groupsDataService;
+            this.clockService = clockService;
+        }
+
+        [Route("{groupId:int}")]
+        public IActionResult Index(
+            int groupId = 1,
+            string? searchString = null,
+            string? sortBy = null,

```

Comment:
There is no sorting in the acceptance criteria so this and `sortDirection` can be dropped.

Reply:
Yes, but we using SearchSortFilterAndPaginate method that doesn't have argument to turn off the sort feature. it, however, can enable search, pagination etc.. Since we reusing the existing delegates search, this will allow users to sort users if they want.

Reply:
This has now been refactored as per the original comment.